src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Diagnostics/CompositionTrace.cs

```diff
@@ -39,8 +51,25 @@ internal static void PartDefinitionRejected(ComposablePartDefinition definition,
 
         internal static void AssemblyLoadFailed(DirectoryCatalog catalog, string fileName, Exception exception)
         {
-            Assumes.NotNull(catalog, exception);
-            Assumes.NotNullOrEmpty(fileName);            
+            if(catalog == null)
+            {
+                throw new ArgumentNullException(nameof(catalog));
+            }
+
+            if(exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+
+            if(fileName == null)
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+
+            if(fileName.Length == 0)
+            {
+                throw new ArgumentException(CommonStrings.ArgumentException_EmptyString, nameof(fileName));
```

Comment:
It would be great if we could also get rid of the CommonStrings file as it only contains two strings. I suggest copying the two strings into the respective projects and change this and other lines to `SR.*` instead of `CommonStrings.*`.